bluetooth: hci: enable `BT_SPI` from DT

JordanYates opened the pull request:
Add dedicated Kconfig symbols for the two SPI backend variants. This
allows enabling the symbols by default from devicetree, and cleans up
the cmake logic.

The `HCI_SPI` symbol is now automatically selected from devicetree,
removing the need for board specific defaults.

JordanYates committed:
bluetooth: hci: dedicated symbols for SPI variants

Add dedicated Kconfig symbols for the two SPI backend variants. This
allows enabling the symbols by default from devicetree, and cleans up
the cmake logic.

Signed-off-by: Jordan Yates <jordan@embeint.com>

jhedberg commented:
Not sure if this is quite correct, due to this hackery in `CMakeLists.txt` (which I'd gladly get rid of, but it may require some refactoring somewhere):
https://github.com/zephyrproject-rtos/zephyr/blob/1aaea30adb62df1aff2ccb167966db443667d5f6/drivers/bluetooth/hci/CMakeLists.txt#L20-L26

JordanYates commented:
> Not sure if this is quite correct, due to this hackery in `CMakeLists.txt` (which I'd gladly get rid of, but it may require some refactoring somewhere):
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/1aaea30adb62df1aff2ccb167966db443667d5f6/drivers/bluetooth/hci/CMakeLists.txt#L20-L26

Yea noticed this after pushing, working on it now

JordanYates committed:
boards: remove `HCI_SPI` defaults

The `HCI_SPI` symbol is now automatically selected from devicetree,
removing the need for board specific defaults.

Signed-off-by: Jordan Yates <jordan@embeint.com>

HoZHel commented on drivers/bluetooth/hci/Kconfig:
```diff
@@ -48,15 +48,46 @@ config BT_HCI_IPC
+	default 75
+
+config BT_SPI_ZEPHYR
+	bool "SPI HCI as implemented by Zephyr"
```
```suggestion
	bool "HCI over SPI as implemented by Zephyr"
```

HoZHel commented on drivers/bluetooth/hci/Kconfig:
```diff
@@ -48,15 +48,46 @@ config BT_HCI_IPC
+	depends on DT_HAS_ZEPHYR_BT_HCI_SPI_ENABLED
+	select BT_SPI
+	help
+	  Bluetooth HCI driver for communication with Bluetooth controllers
```
```suggestion
	  Bluetooth HCI over SPI driver for communication with Bluetooth controllers
```

HoZHel commented on drivers/bluetooth/hci/Kconfig:
```diff
@@ -48,15 +48,46 @@ config BT_HCI_IPC
+	  devices from ST Microelectronics.
+
+config BT_BLUENRG_ACI
+	bool "ACI message with BlueNRG-based devices"
```
```suggestion
	bool "ACI configuration message with BlueNRG-based devices"
```

HoZHel commented on drivers/bluetooth/hci/Kconfig:
```diff
@@ -48,15 +48,46 @@ config BT_HCI_IPC
+	depends on DT_HAS_ST_HCI_SPI_V1_ENABLED || DT_HAS_ST_HCI_SPI_V2_ENABLED
+	select BT_SPI
+	help
+	  Bluetooth HCI driver for communication with the BlueNRG-1 and BlueNRG-2
```
```suggestion
	  Bluetooth HCI over SPI driver for communication with STMicroelectronics radio devices.
```

HoZHel commented on drivers/bluetooth/hci/Kconfig:
```diff
@@ -48,15 +48,46 @@ config BT_HCI_IPC
+	depends on BT_SPI_BLUENRG
+	select BT_HCI_SET_PUBLIC_ADDR
+	help
+	  Enable support for devices compatible with the BlueNRG Bluetooth
+	  Stack. Current driver supports: ST BLUENRG-MS.
```
```suggestion
	  Enable vendor specific commands to switch radio to controller-only mode.
```

JordanYates commented:
@HoZHel the changes in this PR are not modifying the form of any of the Kconfig text, just rearranging the symbols (And duplicating in the case of `SPI_HCI_ZEPHYR`).
I think substantive changes should be their own PR, so that all the symbols can be addressed at once/wording changes discussed independently.

HoZHel commented:
> @HoZHel the changes in this PR are not modifying the form of any of the Kconfig text, just rearranging the symbols (And duplicating in the case of `SPI_HCI_ZEPHYR`). I think substantive changes should be their own PR, so that all the symbols can be addressed at once/wording changes discussed independently.

Well, `config BT_SPI_BLUENRG` is not an rearrangement and it has a wrong definition.

JordanYates commented:
Removed prompts from the new symbols as done in #77418

msmttchr commented:
@JordanYates, recently PR #71352 added a new HCI board which possibly needs update before merging this PR.

JordanYates commented:
> @JordanYates, recently PR #71352 added a new HCI board which possibly needs update before merging this PR.

Done

msmttchr commented:
> > @HoZHel the changes in this PR are not modifying the form of any of the Kconfig text, just rearranging the symbols (And duplicating in the case of `SPI_HCI_ZEPHYR`). I think substantive changes should be their own PR, so that all the symbols can be addressed at once/wording changes discussed independently.
> 
> Well, `config BT_SPI_BLUENRG` is not an rearrangement and it has a wrong definition.

I think this comment needs also to be addressed before merging this PR, since the information provided can be misleading.

msmttchr commented:
> The `BT_SPI_BLUENRG` symbol was updated mostly in line with the request
Ah, sorry I missed that.

JordanYates commented:
Rebasing on main in the hope the unrelated CI failures are resolved

JordanYates commented:
Rebase to resolve release notes conflict

jhedberg commented:
@erwango any idea about the CI failures here? They don't seem to be related to this PR, rather something's off with mbedtls and wifi on ST platforms. Is that a know issue?

erwango commented:
> @erwango any idea about the CI failures here? They don't seem to be related to this PR, rather something's off with mbedtls and wifi on ST platforms. Is that a known issue?

Also failing with `nrf9160dk/nrf9160/ns`, so I guess it's rather an issue with the test itself.

jhedberg commented:
> Also failing with `nrf9160dk/nrf9160/ns`, so I guess it's rather an issue with the test itself.

@erwango where do you see that? I only see `stm32l562e_dk/stm32l562xx/ns` failing in the CI logs.

erwango commented:
> > Also failing with `nrf9160dk/nrf9160/ns`, so I guess it's rather an issue with the test itself.
> 
> @erwango where do you see that? I only see `stm32l562e_dk/stm32l562xx/ns` failing in the CI logs.

Reproduced locally. But I don't know why only `stm32l562e_dk/stm32l562xx/ns` is tested.

jhedberg commented:
The CI failure should get fixed by #80381
